lib/chore/queues/filesystem/publisher.rb

```diff
@@ -10,27 +10,20 @@ class Publisher < Chore::Publisher
         # See the top of FilesystemConsumer for comments on how this works
         include FilesystemQueue
 
-        # Mutex for holding a lock over the files for this queue while they are in process
-        FILE_MUTEX = Mutex.new
-
         # use of mutex and file locking should make this both threadsafe and safe for multiple
         # processes to use the same queue directory simultaneously. 
         def publish(queue_name,job)
           # First try encoding the job to avoid writing empty job files if this fails
           encoded_job = encode_job(job)
 
-          FILE_MUTEX.synchronize do
-            while true
-              # keep trying to get a file with nothing in it meaning we just created it
-              # as opposed to us getting someone else's file that hasn't been processed yet.
-              f = File.open(filename(queue_name, job[:class].to_s), "w")
+          published = false
+          while !published
```

Comment:
i'm sure you thought of this, but i am not intimately familiar with chore, so ...
what are the chances we get stuck in this loop forever?

Reply:
Well, consider how `filename` works.  On each iteration, it's using a combination of the process id and timestamp (to 6 decimal places) to generate a filename.  I can't think of a situation where this loop would infinitely keep generating filenames that conflict with filenames already on the filesystem -- unless time was frozen :)

Reply:
but what about the impact of climate change?
clearly didn't trace the code path :)
LGTM